src/common/string/filter/sequence-matching.ts

```diff
@@ -0,0 +1,65 @@
+import { fuzzyScore, isPatternInWord } from "./filter";
+
+/**
+ * Determine whether a sequence of letters exists in another string,
+ *   in that order, allowing for skipping. Ex: "chdr" exists in "chandelier")
+ *
+ * filter => sequence of letters
+ * word => Word to check for sequence
+ *
+ * returns either:
+ *    number => if word contains sequence, return a score
+ *    string => if word is empty, return the word (to allow for alphabetical sorting of all available words)
+ *    undefined => if no match was found
+ */
+
+export const fuzzySequentialMatch = (filter: string, ...words: string[]) => {
```

Comment:
```suggestion
export const fuzzySequentialMatch = (filter: string, ...words: string[]): number | string | undefined => {
```

Reply:
Typescript is already inferring the function's type (which also includes `| undefined`)

![image](https://user-images.githubusercontent.com/1480827/96473222-38eae300-11e6-11eb-9863-95c3dc6696ad.png)